packages/core/src/tracing/instrumentation/database/prisma.js

```diff
@@ -0,0 +1,204 @@
+/*
+ * (c) Copyright IBM Corp. 2022
+ */
+
+'use strict';
+
+const shimmer = require('shimmer');
+
+let logger;
+logger = require('../../../logger').getLogger('tracing/prisma', newLogger => {
+  logger = newLogger;
+});
+
+const requireHook = require('../../../util/requireHook');
+const { getErrorDetails, getStackTrace } = require('../../tracingUtil');
+const { EXIT } = require('../../constants');
+const cls = require('../../cls');
+
+let isActive = false;
+
+exports.init = function init() {
+  requireHook.onModuleLoad('@prisma/client', instrumentPrismaClient);
+};
+
+function instrumentPrismaClient(prismaClientModule) {
+  instrumentClientConstructor(prismaClientModule);
+  shimRequest(prismaClientModule);
+}
+
+function instrumentClientConstructor(prismaClientModule) {
+  // Additionally instrument the constructor to get access to database URL and type.
+  if (typeof prismaClientModule.PrismaClient === 'function') {
+    class InstanaPrismaClient extends prismaClientModule.PrismaClient {
+      constructor() {
+        super(...arguments);
+
+        // Unfortunately, resolving the configuration is an asynchronous operation.
+        // If the first model access happens in the same event loop iteration as creating the client, the span for that
+        // Prisma operation will not have the provider or target URL available.
+        if (this._engine && typeof this._engine.getConfig === 'function') {
+          const configPromise = this._engine.getConfig();
+          if (typeof configPromise.then === 'function') {
+            configPromise.then(configResult => {
+              if (!configResult || !Array.isArray(configResult.datasources)) {
+                return;
+              }
+              const activeDatasource = configResult.datasources[0];
+              if (!activeDatasource) {
+                return;
+              }
+
+              // We attach the provider and destination URL to the Prisma client instance. That way, when multiple
+              // Prisma client's are used, we do not confuse providers/database URLs
+              this.__inProvider = activeDatasource.activeProvider;
+              const dataSourceUrlObject = activeDatasource.url;
+              if (dataSourceUrlObject && dataSourceUrlObject.value) {
+                this.in_DataSourceUrl = redactPassword(activeDatasource.activeProvider, dataSourceUrlObject.value);
```

Comment:
_Discuss_: Wondering if we should remember the values in a global array for each client, because it might be confusing if somebody is debugging and finds these custom Instana attributes on the PrismaClient class.

Reply:
Good idea. 👍

Reply:
Done